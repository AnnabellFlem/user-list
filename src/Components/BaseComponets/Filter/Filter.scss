@import 'src/Styles/sizes';
@import 'src/Styles/colors';

.filter {
  display: flex;
  justify-content: center;

  &__input {
    width: 100%;
    background: $white-50;
    border: none;
    font-size: $font-s;
    padding: $indent-s $indent-xs;
  }
}
