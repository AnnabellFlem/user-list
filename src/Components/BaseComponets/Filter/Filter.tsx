import React from 'react'
import './Filter.scss'

type FilterProps = {
  input: string
  onChange: any
}

const Filter: React.FC<FilterProps> = ({
  input: keyword,
  onChange: setKeyword,
}) => {
  return (
    <div className="filter">
      <input
        className="filter__input"
        key="key123"
        value={keyword}
        placeholder={'Search user name'}
        onChange={e => setKeyword(e.target.value)}
      />
    </div>
  )
}

export default Filter
