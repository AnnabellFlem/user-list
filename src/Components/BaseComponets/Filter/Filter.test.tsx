import React from 'react'
import { shallow } from 'enzyme'
import Filter from './Filter'

const mockedData = 'data'
const onChangeMock = jest.fn()

describe('Filter test', () => {
  let wrapper: any

  beforeAll(() => {
    wrapper = shallow(<Filter input={mockedData} onChange={onChangeMock} />)
  })

  it('should to match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Component should render div', () => {
    expect(wrapper.find('div')).toHaveLength(1)
  })

  it('Component should render input with data', () => {
    expect(wrapper.find('input')).toHaveLength(1)
    expect(wrapper.find('input').props().value).toBe(mockedData)
  })
})
