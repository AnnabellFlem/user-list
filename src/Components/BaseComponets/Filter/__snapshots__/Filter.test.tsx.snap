// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter test should to match snapshot 1`] = `ShallowWrapper {}`;
